hook.php

```diff
@@ -341,12 +341,37 @@ function plugin_fields_addWhere($link, $nott, $itemtype, $ID, $val, $searchtype)
     /** @var \DBmysql $DB */
     global $DB;
 
-    $searchopt = &Search::getOptions($itemtype);
-    $table     = $searchopt[$ID]['table'];
-    $field     = $searchopt[$ID]['field'];
+    $searchopt    = &Search::getOptions($itemtype);
+    $table        = $searchopt[$ID]['table'];
+    $field        = $searchopt[$ID]['field'];
+    $pfields_type = $searchopt[$ID]['pfields_type'];
 
     $field_field = new PluginFieldsField();
 
+    if (
+        $field_field->getFromDBByCrit(
+            [
+                'name'     => $field
+            ],
+        )
+        && $pfields_type == 'number'
```

Comment:
```suggestion
        $field_field->getFromDBByCrit(
            [
                'name'     => $field,
                'type' => 'number'
            ],
        )
        && $pfields_type == 'number'
```

just an additional control